docs/products/clickhouse/howto/run-federated-queries.rst

```diff
@@ -0,0 +1,159 @@
+
+.. code-block:: bash
+
+   GRANT CREATE TEMPORARY TABLE, S3, URL ON *.* TO <username> [WITH GRANT OPTION]
+
+
+The CREATE TEMPORARY TABLE grant is required for both sources. Adding WITH GRANT OPTION allows the user to further transfer the privileges.
+
+Limitations
+'''''''''''
+
+* Federated queries in Aiven for ClickHouse only support S3-compatible object storage providers for the time being. More external data sources coming soon!
+* Virtual tables are only supported for URL sources, using the URL table engine. Stay tuned for us supporting the S3 table engine in the future!
+
+Run a federated query
+---------------------
+
+Check of the examples of running federated queries to read and pull data from external S3-compatible object storages.
+
+Query using SELECT and the S3 function
+''''''''''''''''''''''''''''''''''''''
+
+SQL SELECT statements using the S3 and URL functions are able to query public resources using the URL of the resource.
+For instance, let's explore the network connectivity measurement data provided by Open Observatory of Network Interference (OONI).
+
+.. code-block:: bash
+
+   WITH ooni_data_sample AS
+      (
+         SELECT *
+         FROM s3('https://ooni-data-eu-fra.s3.eu-central-1.amazonaws.com/clickhouse_export/csv/fastpath_202308.csv.zstd')
+         LIMIT 100000
+      )
+   SELECT
+      probe_cc,
+      test_name,
+      countIf(anomaly = 't') AS num_anomalies
+   FROM ooni_data_sample
+   GROUP BY
+      probe_cc,
+      test_name
+   HAVING num_anomalies > 10
+   ORDER BY num_anomalies DESC
+   LIMIT 50
+
+Query using SELECT and the s3Cluster function
+'''''''''''''''''''''''''''''''''''''''''''''
+
+Using the s3Cluster function allows all the cluster's nodes to participate in the query execution to compute aggregations.
+
+Query a private S3 bucket
+'''''''''''''''''''''''''
+
+Private buckets can be accessed by providing the access token and secret as function parameters.
+
+.. code-block:: bash
+
+   SELECT * 
+   FROM s3('https://private-bucket.s3.eu-west-3.amazonaws.com/dataset-prefix/partition-name.csv','some_aws_access_key_id', 'some_aws_secret_access_key')
+
+Depending on the format, the schema can be automatically detected. If it isn't, you may also provide the column types as function parameters.
+
+.. code-block:: bash
+
+   SELECT * 
+   FROM s3('https://private-bucket.s3.eu-west-3.amazonaws.com/orders-dataset/partition-name.csv',
+   'access_token', 
+   'secret_token', 
+   'CSVWithNames', 
+   "`order_id` UInt64, `quantity` Decimal(9, 18), `order_datetime` DateTime"
+   )
+
+Query using SELECT and the URL function
+'''''''''''''''''''''''''''''''''''''''
+
+.. code-block:: bash
+
+   SELECT *
+   FROM url('https://interesting-public-csv-that-wont-change.some-agency-or-ngo-we-like.org', 'CSVWithNames')
```

Comment:
Missing the actual URL (my bad, forgot to update the document), marking as changes requested so that we don't accidentally publish as-is.

Reply:
@aris-aiven , do you mean we still need to update this query here

Reply:
You could use something like the following
```suggestion
  SELECT
      Year,
      UniqueCarrier,
      Origin,
      Dest,
      sum(toInt32OrZero(LateAircraftDelay)) AS TotalAircraftDelays
  FROM url('https://s3.amazonaws.com/h2o-airlines-unpacked/allyears2k.csv')
  WHERE Year > 2000
  GROUP BY
      Year,
      UniqueCarrier,
      Origin,
      Dest
  ORDER BY TotalAircraftDelays DESC
  LIMIT 20
```

Need to look at how the SQL is rendered and circle back

Reply:
Just to follow up, please don't commit this suggestion, we don't know what the license is. Looking for an alternative with a know Creative Commons license